internal/gatewayapi/helpers.go

```diff
@@ -400,6 +406,10 @@ func irStringKey(gatewayNs, gatewayName string) string {
 	return fmt.Sprintf("%s/%s", gatewayNs, gatewayName)
 }
 
+func irMergedInfraPortName(listener *ListenerContext) string {
```

Comment:
lets flip the order of concatenation, similar to `irHTTPListenerName` ? and we cannot use `-`, we need to use `/` , an invalid char as a delimiter

Reply:
alright, I thought I could use `-` as this is mainly used for container ports, but I will change it to `/` and add this in the infrastructure helper function